compiler/q-implant/src/QImplant.cpp

```diff
@@ -282,4 +285,38 @@ void QImplant::write(loco::Graph *g)
   }
 }
 
+void QImplant::forward_qparam(loco::Graph *g)
+{
+  std::set<luci::CircleOpcode> forwardable_opcode{
+    luci::CircleOpcode::RESHAPE, luci::CircleOpcode::SPLIT, luci::CircleOpcode::TRANSPOSE};
+
+  auto forwardable = [&forwardable_opcode](luci::CircleOpcode opcode) {
+    return forwardable_opcode.find(opcode) != forwardable_opcode.end();
+  };
+
+  for (auto node : loco::postorder_traversal(loco::output_nodes(g)))
+  {
+    auto circle_node = loco::must_cast<luci::CircleNode *>(node);
+    // skip when node is output
+    if (circle_node->opcode() == luci::CircleOpcode::CIRCLEOUTPUT)
+      continue;
+
+    auto quantparam = circle_node->quantparam();
+    if (quantparam == nullptr)
+      continue;
+
+    for (auto successor : loco::succs(node))
+    {
+      auto successor_node = loco::must_cast<luci::CircleNode *>(successor);
+      if (successor_node->quantparam() == nullptr)
+      {
+        if (!forwardable(successor_node->opcode()))
+          continue;
+        assert(circle_node->quantparam());
```

Comment:
plz hoist this line

Reply:
`copy_quantparam` in `CircleQuantParam.cpp` check source quantparam is null or not, so It would unnecessary as you mention.

I will remove it.

Thank you!

Reply:
`circle_node` does not change in this `for (auto successor : loco::succs(node))` loop.
so we can hoist out of this `for` loop.
but then, there is
```cxx
    auto quantparam = circle_node->quantparam();
    if (quantparam == nullptr)
      continue;
```
which, we don't have to do assertion check.

I didn't check with `copy_quantparam`

Reply:
If I understand correctly, you mean that we can do it like
```
if(!circle_node->quantparam())
  continue;
```

Reply:
> If I understand correctly, you mean that we can do it like

I don't know why the code is related with `assert(circle_node->quantparam());` removal.

Reply:
>plz hoist this line this line looks unnecessary

First, I only heard about `hoist` in `javascript`, and not sure about the meaning of sentence.
And when I checked review at `File changed`, the review is attached at `assert` line.
So I understood that assert line is unnecessary.

> circle_node does not change in this for (auto successor : loco::succs(node)) loop.
so we can hoist out of this for loop.

I understood that `quantparam` is only used at `if` condition, so declaring `quantparam` is unnecessary.

> If I understand correctly, you mean that we can do it like

So at here, I asked about (want to check)  `hoist` means that don't have to declare variable and just use it in `if` condition.
I didn't meant that removal assertion or not.

Sorry about lack of my knowledge and English..

Thank you

Reply:
> I asked about (want to check) hoist means

FYI https://en.wikipedia.org/wiki/Loop-invariant_code_motion

Reply:
Now I got what you meant.

`assert(circle_node->quantparam());` is not related to successor_node, so need to check it not in side of successor loop but in side of node loop.

> but then, there is

But we have already checked quantparam, so there is no need to check it. so you strikethrough the `hoist` sentence.
I knew about loop-invariant conditions, but don't know about `hoist` means.
So when you say the line is unnecessary, I start to check near code and when checking `copy_quantparam`, it checks `qparam` so I just thought `unnecessary` is checking qparam even the function process well. 

Sorry about lack of conversation.

Thank you